jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1488,13 +1489,33 @@ Method getMethodHelper(
 			return null;
 		}
 	}
-	result = forDeclaredMethod ? getDeclaredMethodImpl(name, parameterTypes, strSig, null) : getMethodImpl(name, parameterTypes, strSig);
+	
+	if (forDeclaredMethod) {
+		result = getDeclaredMethodImpl(name, parameterTypes, strSig, null);
+	} else {
+		result = getMethodImpl(name, parameterTypes, strSig);
+		/* Retrieve the specified method implemented by the superclass from the top to the bottom
+		 * Note: there is no need do so when the method is declared by the current class.
+		 */
+		if ((result != null) && result.getDeclaringClass().isInterface() && (this != Object.class)) {
```

Comment:
Is the check `this != Object.class` necessary?

Reply:
I believe this is referring to the javadoc line: 
> If C is a class other than Object, then include the result of invoking this algorithm recursively on the superclass of C.
Though in practice I think it may not be needed. @ChengJin01 do you have any insights to this from your testing?

Reply:
It looks like this check is also done https://github.com/eclipse/openj9/blob/8bf4dc691e37ba4c7cfc0ebe85f78e8c12736625/jcl/src/java.base/share/classes/java/lang/Class.java#L1594 so its not necessary to do it twice.

Reply:
@theresa-m, `Object.class` should be excluded in the search of superclass in any case. Please modify accordingly if any duplicate and verify the update with all tests.